apps/backend/src/groups/groups.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { Body, Controller, Delete, Get, Param, Patch, Post } from '@nestjs/common';
+import { Prisma, Role } from '@prisma/client';
+
+import { GroupsService } from './groups.service';
+
+@Controller('groups')
+export class GroupsController {
+  constructor(private readonly groupsService: GroupsService) {}
+
+  @Post()
+  create(@Body() data: Prisma.GroupCreateInput) {
+    return this.groupsService.create(data);
+  }
+
+  @Get()
+  findAll() {
+    return this.groupsService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.groupsService.findOne(Number(id));
+  }
+
+  @Get(':id/members')
+  findMembers(@Param('id') id: string) {
+    return this.groupsService.findMembers(Number(id));
+  }
+
+  @Patch(':id')
+  update(@Param('id') id: string, @Body() data: Prisma.GroupUpdateInput) {
+    return this.groupsService.update(Number(id), data);
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.groupsService.remove(Number(id));
+  }
+
+  @Post(':id/members/:userId')
+  addMember(@Param('id') id: string, @Param('userId') userId: string) {
+    return this.groupsService.addMember(Number(id), Number(userId));
+  }
+
+  @Patch(':id/members/:userId')
+  updateRole(@Param('id') id: string, @Param('userId') userId: string, @Body() newRole: Role) {
```

Comment:
I suspect the body parameter should look something like newRole: { role: Role }, because of JSON, or did it work for you in that form?

Reply:
Tesztelni sajnos nem tudtam mivel a user része még nincs implementálva a branchen.

Reply:
Mégis tudom tesztelni, de sajnos az enumot sehogy se tudom rendes átvenni. Talán úgy kéne hogy:
{ newRole: 0 } lenne az ADMIN az enum miatt de úgy se tudok rácheckolni if (newRole === Role.ADMIN). Úgyhogy ebben most így megakadtam, de a többit kijavítottam és commitelem is.

Reply:
Tesztelni úgy tudod, hogy az adatbázist manuálisan feltöltöd user adatokkal (`prisma studio`), de erre lehet te is rájöttél. Nekem `newRole: { role: Role }` működött. Az enum validációval most itt nem kell foglalkoznod, a prizma úgyis exceptiont dob, ha más adat érkezik. Ezzel a paraméterezéssel a bodyban egy olyan objektumot kell adni, aminek egy role key-e van, és a service-ben newRole.role-ként éred el

Reply:
Sehogy sem sikerült nekem olyan JSON bodyt írnom, hogy checkelni tudjam. Mindig egy ilyen objectet kaptam: (kép).
![Képernyőkép 2024-05-28 125051](https://github.com/kir-dev/collage-schedule-planner/assets/117939375/25104dd5-c6de-4054-a6a5-42d494207f68)
Pedig már azt is megpróbáltam, hogy a newRole string típusú, de úgy is ugyanaz volt a helyzet.
Még arra gondoltam, hogy ezt a műveletet szét lehetne szedni, két külön endpointra. Lehetne egy `<group_id>/members/<user_id>/promote` meg ennek a párja a `<group_id>/members/<user_id>/demote`. Ennek ugye az lenne a hátránya, hogy csak ezzel a két rolelal működik és tovább nem bővíthető. Szerinted melyik lenne a jobb megoldás?